src/UI/Conversation.jsx

```diff
@@ -146,33 +155,58 @@ class Conversation extends Component {
 				<div className={styles.body}>
 					{
 						this.state.welcomeMessage
-							&& <Announcement message={this.state.welcomeMessage} />
+						&& <Announcement message={this.state.welcomeMessage} />
 					}
 					{
 						this.state.messages.map((message, index, array) => (
 							<React.Fragment key={message.id}>
 								{
 									this.setRenderedDate(this.getDateFromTimestamp(message.time))
-										&& <DateGroup timestamp={message.time} />
+									&& <DateGroup timestamp={message.time} />
+								}
+								{
+									message.carousel.length === 0
+									&& <Message
+										api={this.props.api}
+										message={message}
+										showAgent={this.shouldRenderAgentName(index, index - 1)}
+									   />
+								}
+								{
+									message.carousel && message.carousel.length > 0
+									&& <Carousel
+										items={
+											message.carousel.map((carouselItem, _index) => (
```

Comment:
Ik snap niet hoe dit werkt, je mapt over de data van carousel en dan geef je het message object terug aan carousel?
Als ik het goed begrijp returnt hij een nieuwe array van message classes die je dan toont in carousel

Reply:
Wat dit doet is 1 Carousel maken en daar geeft ie `items` aan mee.
De `items` wordt gevuld met alle carousel "items" uit de `message.carousel`.
Voor elke carousel "item" bouwen we een message component op.
Een message component heeft een aantal dingen nodig, zoals `agent`, `id`, `media`, `time` welke NIET in `message.carousel` zitten. Vandaar de `...message`.
Daarnaast geven we carousel specifieke dingen mee aan de message zoals `title`, `message`, `media` en `buttons`.

Je eindigt dus met 1 message voor elke carousel "item". Al deze messages worden in een Carousel component gezet zodat je kan "scrollen".

Reply:
Yes snap m! het `...message` stuk dat niet allemaal in carousel zat was me zeker niet duidelijk, maar nu wel thanks!